internal/providercache/installer.go

```diff
@@ -41,8 +42,17 @@ type Installer struct {
 	// namespace, which we use for providers that are built in to Terraform
 	// and thus do not need any separate installation step.
 	builtInProviderTypes []string
+
+	// pluginProtocolVersion is the protocol version terrafrom core supports to
+	// communicate with servers, and is used to resolve plugin discovery with
+	// terraform registry, in addition to any specified plugin version
+	// constraints.
+	pluginProtocolVersion string
 }
 
+// The currently-supported plugin protocol version.
+const PluginProtocolVersion = "5"
```

Comment:
I noticed that we used `uint` instead of string in `plugins/discovery/`, but also that it (eventually) got parsed as a string anyway, so that's what I went with here. I'm quite open to changing that.

Reply:
It looks like all of the uses of this immediately parse it into a version constraints object anyway... if that's true, perhaps it would be clearer to make this be a version constraints object immediately, at startup:

```go
const SupportedPluginProtocols = getproviders.MustParseVersionConstraints("5")
```

(incidentally, I'd prefer to use the `getproviders` wrappers around the version and constraint parsers, rather than calling directly into the underlying versions library, because that way we can centralize decisions like the one to parse in "ruby style" rather than "npm style".)